doc/develop/api/api_conventions.rst

```diff
@@ -0,0 +1,102 @@
+.. _api_conventions.rst:
+
+API Conventions
+###############
+
+Zephyr provides a few programming interfaces that users can use for creating their applications.
+Zephyr's API space includes the following areas:
+- C programming interface (function prototypes, structures, and macros), provided in C header files
+- Configuration system (Kconfig)
+- Device driver model (Devicetree)
+
+Other areas, such as the build system (CMake), may also be seen as Zephyr's API. However, any area
```

Comment:
How about zephyr specific CMake commands like `zephyr_library()` and `zephyr_library_sources_ifdef`? These are zephyr specific APIs right?

Reply:
Yes these are an API. However, we decided to limit the scope to items listed above. The motivation is to keep the scope narrow for now to make some progress and see if the policy works. We will improve over time.